vdaas/vald
# Update makefile for "not implemented" placeholder

### Description:

This PR update `make gotests/gen` command to generate test implementation with `not implemented below` placeholder.
The placeholder indicate that the test implementation below the placeholder is not implemented yet. Everytime the `make gotests/gen` command runs, the test implementation below the placeholder will be removed and the new test implementation will be generated below the placeholder.

It ensures that the not implemented test code will keep updated with the latest test template. 

I have updated the `make gotests/gen` command to execute the following steps:

1. Create test file with package definition if the test file does not exists
2. find and remove the content of the placeholder and the content below the placeholder
3. concat the placeholder to `*_test.go` file
4. execute gotests tool to generate test files
5. remove test file with no implementation to avoid empty test file generated
6. execute gotests/patch to patch go import path
7. execute format/test to format go test files

To confirm it is working, 
1. execute `make gotests/gen` on your local environment.
2. Validate changes on test files, including the placeholder `// NOT IMPLEMENTED BELOW` is append to the correct place, and test stub generated or not, and if test file does not exists, the file will be generated.
3. To ensure no extra changes applied when re-executing, execute the command again, and validate if anything changes from step 2

### Related Issue:






### Versions:



- Go Version: 1.20.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 2.0.9

### Checklist:




- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### Special notes for your reviewer:
